site/src/pages/WorkspacePage/WorkspaceActions/constants.ts

```diff
@@ -1,31 +1,27 @@
-import { Workspace, WorkspaceStatus } from "api/typesGenerated";
-import { ReactNode } from "react";
+import { type Workspace, type WorkspaceStatus } from "api/typesGenerated";
+import { type ReactElement } from "react";
 
-// the button types we have
-export enum ButtonTypesEnum {
-  start = "start",
-  starting = "starting",
-  stop = "stop",
-  stopping = "stopping",
-  restart = "restart",
-  restarting = "restarting",
-  deleting = "deleting",
-  update = "update",
-  updating = "updating",
-  activate = "activate",
-  activating = "activating",
-  // disabled buttons
-  canceling = "canceling",
-  deleted = "deleted",
-  pending = "pending",
-}
+/**
+ * Buttons supported by workspace actions. Canceling, Deleted, and Pending
+ * should all be associated with disabled states
+ */
+export type ButtonType =
```

Comment:
Wanted to get rid of the enum, just because they can have weird edge case behavior at runtime. It also felt like one of the benefits was making it more clear that the majority of these types are synced with the WorkspaceStatus type